compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeParamAnalyzer.java

```diff
@@ -475,7 +475,8 @@ private void findTypeParamInTuple(Location loc, BTupleType expType, BTupleType a
     private void findTypeParamInStream(Location loc, BStreamType expType, BStreamType actualType,
                                        SymbolEnv env, HashSet<BType> resolvedTypes, FindTypeParamResult result) {
         findTypeParam(loc, expType.constraint, actualType.constraint, env, resolvedTypes, result);
-        findTypeParam(loc, expType.error, actualType.error, env, resolvedTypes, result);
+        findTypeParam(loc, expType.error, (actualType.error != null) ? actualType.error : symTable.nilType, env,
```

Comment:
now that the default value of stream type error is set to nil, `(actualType.error != null)` is redundant